profiles/common/modules/features/nexteuropa_newsroom/nexteuropa_newsroom.multilingual.inc

```diff
@@ -0,0 +1,701 @@
+ */
+function _nexteuropa_newsroom_prepare_dynamic_mapping(array $premapping, $type = NULL) {
+  $def_lang = drupal_strtoupper(language_default()->language);
+  // Get enabled languages codes.
+  $lang_keys = _nexteuropa_newsroom_languages();
+  // Loop through all the fields, build the needed arrays.
+  $i = 0;
+
+  foreach ($premapping as $field => $values) {
+    if (isset($values['tamper'])) {
+      $tamper = _nexteuropa_newsroom_prepare_dynamic_tamper($values['tamper'], $i, $type);
+      _nexteuropa_newsroom_dynamic_tamper($tamper);
+    }
+
+    // Things look different if the field is translatable.
+    if ($values['language']) {
+      // If fields are translatable,
+      // we create a different mapping for each language.
+      foreach ($lang_keys as $lang_key) {
+        $upper_key = drupal_strtoupper($lang_key);
+        // Xpath queries.
+        $queries['xpathparser:' . $i] = $values['query'] . '[@lang="' . $upper_key . '"]/text()';
+
+        if (strpos($type, 'topic') && $upper_key != $def_lang) {
+          $queries['xpathparser:' . $i] = '//channel/item[infsonewsroom:BasicSvType="Newsroom service"]/category[@domain!="Newsletter" and @lang="' . $upper_key . '" and @infsonewsroom:TopicId="$Temporary target 1"]/' . $values['query'];
+        }
+        elseif (strpos($type, 'topic') && $upper_key == $def_lang) {
+          $queries['xpathparser:' . $i] = $values['query'];
+        }
+        elseif (strpos($type, 'newsroom_types')) {
+          $queries['xpathparser:' . $i] = $values['query'] . '[@lang="' . $upper_key . '"]/text()';
+        }
+        // Mapping arrays.
+        $mapping[$i] = array(
+          'source' => 'xpathparser:' . $i,
+          'target' => $field . ':et:' . $lang_key,
+          'unique' => FALSE,
+        );
+        // Additional values.
+        if (isset($values['options'])) {
+          foreach ($values['options'] as $key => $option) {
+            $mapping[$i][$key] = $option;
+          }
+        }
+        // Guid is the "unique" field but let's do this "dinamically".
+        if (isset($values['unique'])) {
+          $mapping[$i]['unique'] = TRUE;
+        }
+        $i++;
+      }
+    }
+    // Not translatable fields.
+    else {
+      // Xpath queries.
+      $queries['xpathparser:' . $i] = $values['query'];
+      // Mapping arrays.
+      $mapping[$i] = array(
+        'source' => 'xpathparser:' . $i,
+        'target' => $field,
+        'unique' => FALSE,
+      );
+
+      if (isset($values['options'])) {
+        foreach ($values['options'] as $key => $option) {
+          $mapping[$i][$key] = $option;
+        }
+      }
+      if (isset($values['unique'])) {
+        $mapping[$i]['unique'] = TRUE;
+      }
+      $i++;
+    }
+  }
+
+  // Build also the debug and raw_xml arrays.
+  foreach ($queries as $xpath => $query) {
+    $debug[$xpath] = $raw_xml[$xpath] = 0;
+  }
+
+  return array(
+    'mapping' => $mapping,
+    'queries' => $queries,
+    'debug' => $debug,
```

Comment:
please put debug related functionalities somewhere else, during a normal rendering you don't need them. it's a good practice to always generate just the minimal not the maximum what you actually don't need always.